library/Rules/Tld.php

```diff
@@ -121,6 +121,12 @@ class Tld extends AbstractRule
 
     public function validate($input)
     {
-        return in_array(strtolower($input), $this->tldList);
+        // prefere LayerShifter\TLDDatabase if available
+        if (class_exists('\LayerShifter\TLDDatabase\Store')) {
+            $tldDb = new \LayerShifter\TLDDatabase\Store();
+            return $tldDb->isExists(strtolower($input));
```

Comment:
If we're get to use `mb_strtoupper` twice may be a good idea to have a variable for that

Reply:
In my opinion strtolower will only be called once, because of the if->return statement. If we use a variable for that, it will consume storage.